homeassistant/components/roku/config_flow.py

```diff
@@ -0,0 +1,128 @@
+"""Config flow for Roku."""
+from typing import Any, Dict, Optional
+from urllib.parse import urlparse
+
+from requests.exceptions import RequestException
+from roku import Roku, RokuException
+import voluptuous as vol
+
+from homeassistant.components.ssdp import (
+    ATTR_SSDP_LOCATION,
+    ATTR_UPNP_FRIENDLY_NAME,
+    ATTR_UPNP_SERIAL,
+)
+from homeassistant.config_entries import CONN_CLASS_LOCAL_POLL, ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
+
+ERROR_CANNOT_CONNECT = "cannot_connect"
+ERROR_UNKNOWN = "unknown"
+
+
+def validate_input(data: Dict) -> Dict:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    roku = Roku(data["host"])
+    device_info = roku.device_info
+
+    return {
+        "title": data["host"],
+        "host": data["host"],
+        "serial_num": device_info.serial_num,
+    }
+
+
+class RokuConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a Roku config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_POLL
+
+    @callback
+    def _show_form(self, errors: Optional[Dict] = None) -> Dict[str, Any]:
+        """Show the form to the user."""
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors or {},
+        )
+
+    async def async_step_import(
+        self, user_input: Optional[Dict] = None
+    ) -> Dict[str, Any]:
+        """Handle configuration by yaml file."""
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(
+        self, user_input: Optional[Dict] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initialized by the user."""
+        if not user_input:
+            return self._show_form()
+
+        errors = {}
+
+        try:
+            info = await self.hass.async_add_executor_job(validate_input, user_input)
+        except (RequestException, RokuException):
+            errors["base"] = ERROR_CANNOT_CONNECT
+            return self._show_form(errors)
+        except Exception:  # pylint: disable=broad-except
+            return self.async_abort(reason=ERROR_UNKNOWN)
```

Comment:
Make sure you log the exception as it's unexpected.
```suggestion
            _LOGGER.exception("Unknown error trying to connect.")
            return self.async_abort(reason=ERROR_UNKNOWN)
```